sunpy/net/scraper.py

```diff
@@ -80,28 +80,31 @@ class Scraper:
     pattern looks with the actual time.
```

Comment:
No sure if for this PR possibly a separate doc PR but I think one of the main issues has aways been a clear description of the scraper algorithm so something like.

1. Given a pattern p the smallest unit of time (SUoT) is derived by xxx
2. For each time between start and stop in units of SUoT the pattern is filled.
3. The location given by the filled pattern is visited and a list of files at the location is obtained
4. The name of each file is examined to determine if it matches the remaining pattern
5. ....

Ideally with an/some examples and links to relevant functions

Reply:
I agree, we should add this, maybe as a topic guide?

Reply:
Yes, I think documentation about how the scraper works / the algorithm would be great to have too, do you think I should I add it in this PR itself or a separate one?

Reply:
Do a separate one

Reply:
Also, I should mention this here as well, the doc PR is [here](https://github.com/exitflynn/sunpy/pull/2)